### ISSUE TITLE
Lag during put_rectext, two functions running can almost freeze the game

### ISSUE THREAD

**Issue by Toma400** [Toma400/The_Isle_of_Ansur#51]
As in title, `put_rectext` (introduced in 8bfec43) in its current implementation lags the game heavily. Two heavy nested loops put into every program frame are obviously hitting the gameplay, and even though after optimisation (6b0c524) making one function running bearable, this should not stay, especially as someone may want to add new functions through scripts.

Concepts which can help:

- Making frame counter and designing such heavy features to run based on that, which will decrease amount of re-looping (instead of 20 frames/s, it will run rarer)
- Optimising code further, either by removal of loops or checking for possible aspects that do not need to run simultaneously

Materials which can help:

- [CodeProject general PyGame optimisation list](https://www.codeproject.com/Articles/5298051/Improving-Performance-in-Pygame-Speed-Up-Your-Game)
- [Google search with quite helpful links](https://bit.ly/3DFMAE1)

This is also kinda "call to arms" towards other aspects of the system needing optimisation, as there can be another reasons why the game lags in the first place - though `put_rectext` function made it clear and obvious, so this is the priority.

**Comment by Toma400** [comment]
This goes pretty far into optimalisation part of the game, but also into rewrite of procedural programming part of IoA into OOP one.
Since `put_rectext` isn't used widely, it isn't critical issue to address right now, and it will be automatically solved by introduction of Text object and one-time load.

Moving into pre-alpha 4.

**Comment by Toma400** [comment]
Additionally, there's interesting issue with the same element, so passing info here:
> Possible issue with - https://github.com/Toma400/The_Isle_of_Ansur/blob/6b0c524d4f841b8b19c9503f956885b8c78d0416/core/graphics/text_manag.py - rect/text spacing in put_rectext

**Comment by Toma400** [comment]
Pushing this far beyond current scope. PA7 we go!

**Comment by Toma400** [comment]
Made it into official PA7 scope - but, it'd be good to check if this function is even used currently. I was fairly sure it could've been replaced by PyGame_GUI and through GitHub search it looks like it.
If that's the case, put @Deprecated decorator on this function; if that's not the case, see where function is used and try replacing it by either PyGame_GUI or `put_text`.
Once that is done, close this issue - potential removal of @Deprecated funcs can be handled by separate issue that will handle code cleaning later down the line.
